runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java

```diff
@@ -320,27 +469,155 @@ public void onGameStateChanged(GameStateChanged event)
 		checkStateChanged(false);
 	}
 
+	void resetPingMenu()
+	{
+		advancedPingMenuOpen = false;
+		mouseStartPosition = null;
+		pendingPartyPing = null;
+	}
+
+	private boolean pingIsSameSourceAndTarget(PartyPingData data, PartyPingData other)
+	{
+		if (data.getSourcePlayerIdx() != other.getSourcePlayerIdx() && data.getTargetType() != other.getTargetType())
+		{
+			return false;
+		}
+
+		if (data.getPoint() == null)
+		{
+			return other.getPoint() == null && data.getTargetActor() == other.getTargetActor() && data.getTargetObject() == other.getTargetObject();
+		}
+
+		return data.getPoint().equals(other.getPoint());
+	}
+
 	@Subscribe
-	public void onTilePing(TilePing event)
+	public void onPartyPing(PartyPing ping)
 	{
-		if (config.pings())
+		if (ping.getWorld() != client.getWorld() || !config.pings())
+		{
+			return;
+		}
+		
+		PartyPingTargetType targetType = ping.getTargetType();
+		if (targetType != PartyPingTargetType.NPC && (ping.getPoint().getPlane() != client.getPlane() || !WorldPoint.isInScene(client, ping.getPoint().getX(), ping.getPoint().getY())))
+		{
+			return;
+		}
+
+		Integer target = ping.getTarget();
+		Color color = null;
```

Comment:
This is set to some value (null or user preference) later, it does not need to be initialized here.
```suggestion
		Color color;
```